packages/oidc-client/src/main.ts

```diff
@@ -0,0 +1,83 @@
+import * as core from '@actions/core'
+import * as actions_http_client from '@actions/http-client'
+import {
+  createHttpClient,
+  isSuccessStatusCode,
+  getApiVersion
+} from './internal/utils'
+import jwt_decode from 'jwt-decode'
+import {getIDTokenFromEnv, getIDTokenUrl} from './internal/config-variables'
+
+export async function getIDToken(audience: string): Promise<string> {
+  try {
+    //Check if id token is stored in environment variable
+
+    let id_token: string = getIDTokenFromEnv()
+    if (id_token !== undefined && id_token !== '') {
+      const secondsSinceEpoch = Math.round(Date.now() / 1000)
+      const id_token_json: any = jwt_decode(id_token)
+      if ('exp' in id_token_json) {
+        if (id_token_json['exp'] - secondsSinceEpoch > 300) {
```

Comment:
This value should be less than 5 mins. We initially thought to keep it as 2 mins. But it should be based on some telemetric data as mentioned by @navin22 .